### DISCUSSION TITLE
When OpenFile is in directory mode, should filters apply?

### DISCUSSION THREAD

**Discussion by ptomato** [flatpak/xdg-desktop-portal#1442] [Questions]
Reading the [OpenFile spec](https://flatpak.github.io/xdg-desktop-portal/docs/doc-org.freedesktop.portal.FileChooser.html#org-freedesktop-portal-filechooser-openfile) I would assume that I should be able to specify a filter, e.g. `filter: [('RTFD documents', [(0, '*.rtfd')])]` and have it apply to folder chooser mode (`directory: true`).

However I heard [on the Nautilus issue tracker](https://gitlab.gnome.org/GNOME/nautilus/-/issues/3591) that that is a surprising interpretation of the spec.

The spec itself doesn't seem to clarify either way. It would be good to specify this.

The use case is for apps whose conceptual unit of a "document" is a folder. The best-known example is probably [RTFD documents](https://en.wikipedia.org/wiki/Rich_Text_Format_Directory). My use case specifically is for [Inform](https://flathub.org/apps/com.inform7.IDE), a development environment that stores its projects in folders named `*.inform`.

This kind of document is more common on MacOS (where it is [transparently supported by the OS](https://en.wikipedia.org/wiki/Bundle_(macOS))) and indeed the Inform folder format originated there, but it would be good for interoperability and software freedom to provide facilities for dealing with these kinds of folders in the Portal spec.

Slightly related is https://github.com/flatpak/xdg-desktop-portal/discussions/1419.

**Comment by Mikenux** [comment]
In the spec, about filter options: "List of serialized **file** filters". If you see "file", it implicitly means that these are not folders.

However, we may prefer to treat these folders as files, as in macOS. However, it is necessary to know how to distinguish a folder with an extension that is a file from a folder with an extension that is a folder. Visually, this seems easy using icons, for example. In terms of accessibility for people who can't rely on visual information, I don't know. It is also a concept that is best adopted across desktop environments.

**Reply by Mikenux** [reply]
> In the spec, about filter options: "List of serialized **file** filters". If you see "file", it implicitly means that these are not folders.

However, of course, the documentation can be more explicit.
